### PR TITLE
core: transition to c++17, deboostification

### PR DESCRIPTION
**Pull request by kittywhiskers**

### CODE REVIEW

**Review by PastaPastaPasta — DISMISSED** [review]
utACK for merging via merge commit

For 21064, can we do https://github.com/bitcoin/bitcoin/pull/21064/files#diff-dac26d1a1b4a405373f40090c92278a64cbda24c875d3394d5e7f81747367702L70-L71? git grep doesn't yield any results

Also, does it make sense to remove the cxx17 ci build now?

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
1. agree with @PastaPastaPasta - we should drop `linux64_cxx17` build (should we add `--with-sanitizers=undefined` to `linux64` build?)
2. should apply 20471 to `depends/packages/protobuf.mk`

**Comment by kittywhiskers** [comment]
@PastaPastaPasta, we could [remove those two lines from the inter whitelist](https://github.com/bitcoin/bitcoin/pull/21064/files#diff-dac26d1a1b4a405373f40090c92278a64cbda24c875d3394d5e7f81747367702L70-L71) but they're not there to begin with so there's nothing to remove (which is kinda surprising but not too problematic)

@UdjinM6, #20471 has been updated and `linux64_cxx17` has been dropped

**Review by UdjinM6 — APPROVED** [review]
utACK

File: ci/matrix.sh [hunk 1]
```diff
@@ -59,13 +59,6 @@ elif [ "$BUILD_TARGET" = "linux64" ]; then
   export BITCOIN_CONFIG="--enable-zmq --enable-glibc-back-compat --enable-reduce-exports --enable-crash-hooks"
   export CPPFLAGS="-DDEBUG_LOCKORDER -DENABLE_DASH_DEBUG -DARENA_DEBUG"
   export PYZMQ=true
-elif [ "$BUILD_TARGET" = "linux64_cxx17" ]; then
-  export HOST=x86_64-unknown-linux-gnu
-  export DEP_OPTS="NO_UPNP=1 DEBUG=1"
-  export BITCOIN_CONFIG="--enable-zmq --enable-glibc-back-compat --enable-reduce-exports --enable-crash-hooks --enable-c++17 --enable-werror --with-sanitizers=undefined"
```
Comment (PastaPastaPasta) [inline ci/matrix.sh:65]: Hmmmm, we shouldn't just do this b/c of `--enable-werror --with-sanitizers=undefined` which are not on linux64/32

Also the fact that this one doesn't do functional tests. We should probably keep this, but just rename it to be like linux64_nofun :D

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Comment by PastaPastaPasta** [comment]
(Otherwise everything does good)

**Comment by UdjinM6** [comment]
CI is broken:
- linux32: looks like smth that could be fixed by #4398
- linux64: we probably hit some gitlab limit there `ERROR: Uploading artifacts as "archive" to coordinator... too large archive`

Let's maybe drop 63f0fc4 for now and implement it in a separate PR?

**Comment by PastaPastaPasta** [comment]
> Let's maybe drop [63f0fc4](https://github.com/dashpay/dash/commit/63f0fc4914430abcf03e3f399c7378780f303ad4) for now and implement it in a separate PR?

If we drop 63f0fc4, we should drop https://github.com/kittywhiskers/dash/commit/73de8a61994a993477a0f7b73932dac68939686d as well

**Comment by kittywhiskers** [comment]
Updated

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — APPROVED** [review]
Not a huge fan of merging PRs with failed CI builds but these failures are unrelated, so

utACK

**Pull request merged**